framework/auth/tasks.py

```diff
@@ -59,7 +59,7 @@ def update_affiliation_for_orcid_sso_users(user_id, orcid_id):
         logger.info(f'Eligible institution affiliation has been found for ORCiD SSO user: '
                     f'institution=[{institution._id}], user=[{user_id}], orcid_id=[{orcid_id}]')
         if not user.is_affiliated_with_institution(institution):
-            user.affiliated_institutions.add(institution)
+            InstitutionAffiliation.create(user, institution)
```

Comment:
ORCiD is a special case of SSO, should I use verified ORCiD as SSO identity?

Reply:
fixed in https://github.com/CenterForOpenScience/osf.io/pull/10236/commits/eb013d7b85a6f9fe23de4c5a118d6b05414a37e4